src/pages/product-detail/product-detail.js

```diff
@@ -0,0 +1,130 @@
+                                <p><b>{spec.spec_name}:</b> {spec.spec_value}</p>
+                            )
+                        })
+                    }
+                </div>
+            )
+        }
+    }
+
+    const showCategory = (categoryId) => {
+        var category = categories.results.filter((category)=>{ return category.id === categoryId})
+        if(category.length>0){
+            return(
+                <p><b>Category:</b> {category[0].data.name}</p>
+            )
+        }
+    }
+
+    const showImages = (images) => {
+        if(images != undefined && images.length>0){
+            return(
+                <Swiper
+                    navigation={true} 
+                    modules={[Navigation]} 
+                    className="mySwiper"
+                >
+                {
+                    images.map((imageWrapper, index)=>{
+                        return(
+                            <SwiperSlide>
+                                <img className='swipeObject' src={imageWrapper.image.url}/>
+                            </SwiperSlide>
+                        )
+                    })
+                }
+            </Swiper>
+            )
+        }
+    }
+
+    const showDescription = (description) => {
+        if(description != undefined && description.length>0){
+            return(
+                <div>
+                    <h5>About this product:</h5>
+                    <p>{description[0].text}</p>
+                </div>
+            )
+        }
+    }
+
+    const showAddCartButton = (title, price) => {
+        return(
+            <div className='productTitle'>
+                <h1>{title}</h1>
+                <h3 className='price'>$ {price}</h3>
+                <input type="number"/>
+                <button className="addToCart">Add to cart</button>
+            </div>
+        )
+    }
+
+    const showInfo = () => {
+        if(detailInfo.results != undefined && detailInfo.results.length>0){
+            let item = detailInfo.results[0]
+            
+            return(
+                <div>
+                    {showImages(item.data.images)}
+                    {showAddCartButton(item.data.name, item.data.price)}
+                    <div>
+                        <div className='contentDetailLeft'>
+                            {showDescription(item.data.description)}
+                        </div>
+                        <div className='contentDetailRight'>
+                            <h5>Product details:</h5>
+                            {showCategory(item.data.category.id)}
+                            {showTags(item.tags)}
+                            <p><b>SKU:</b> {item.data.sku}</p>
+                            {showSpecs(item.data.specs)}
+                        </div>
+                    </div>
+                </div>
+            )
+        }
+    }
+
+    return (
+        <div className='ProductDetail'>
+            {showInfo()}
```

Comment:
The way this component is defined is hard to understand because it is just a call to a function that calls other functions to get the HTML, a way to improve the definition could be to move the conde inside the functions to the return section, this way it is evident how the HTML looks like and it is not obfuscated with function calls 

The definition of most components looks something like:

```jsx
function ComponentName({ prop1, prop2, ... }) {
  // STATE
  const [state, setState] = useState();

  ...
  // HANDLERS
  const handleClick = () => setState(something);
  
  ...
  // EFFECTS
  useEffect(() => {
    doSomething();
  }, [someDeps]);

  return (
    // HTML Markup
  );
}